### PR TITLE
Fix sndio support

### PR DESCRIPTION
**Pull request by lxeiqr**
This patch allows sandboxed programs to access a sndiod server and play audio on systems where sndio handles audio.

### CODE REVIEW

**Review by SkewedZeppelin — CHANGES_REQUESTED** [review]
-1, don't add /tmp to this, will effectively apply private-tmp to all profiles using whitelist-common

**Comment by SkewedZeppelin** [comment]
This is likely something you should add to your locals.

**Review by rusty-snake — CHANGES_REQUESTED** [review]
- [ ] Move it into the common block (the first one)
- [x] Only `whitelist` in `${HOME}` (if `whitelist /tmp/sndio` is necessary, we should add it to `private-tmp`)

**Comment by lxeiqr** [comment]
>     * [ ]  Move it into the common block (the first one)
> 
>     * [ ]  Only `whitelist` in `${HOME}` (if `whitelist /tmp/sndio` is necessary, we should add it to `private-tmp`)
I've removed that last line and moved the cookie whitelist into the common block. However, I'm not quite sure how to handle the /tmp/sndio directory. Usually it contains UNIX domain sockets (e.g. /tmp/sndio/sndio0) used to connect to the server.

File: etc/inc/whitelist-common.inc [hunk 1]
```diff
@@ -24,6 +24,7 @@ whitelist ${HOME}/.local/share/icons
 whitelist ${HOME}/.local/share/mime
 whitelist ${HOME}/.mime.types
 whitelist ${HOME}/.uim.d
+whitelist ${HOME}/.sndio/cookie
```
Comment (rusty-snake) [inline etc/inc/whitelist-common.inc:27]:
```suggestion
whitelist ${HOME}/.sndio/cookie
whitelist ${HOME}/.uim.d
```

**Comment by rusty-snake** [comment]
> However, I'm not quite sure how to handle the /tmp/sndio directory. Usually it contains UNIX domain sockets (e.g. /tmp/sndio/sndio0) used to connect to the server.

I think a `profile_add("whitelist /tmp/sndio");` in `fs_private_tmp` works

https://github.com/netblue30/firejail/blob/447e30e2c9b24f4ec6af5520a37a77068e66383b/src/firejail/fs.c#L1251-L1299

**Comment by lxeiqr** [comment]
> > However, I'm not quite sure how to handle the /tmp/sndio directory. Usually it contains UNIX domain sockets (e.g. /tmp/sndio/sndio0) used to connect to the server.
> 
> I think a `profile_add("whitelist /tmp/sndio");` in `fs_private_tmp` works
> 
> https://github.com/netblue30/firejail/blob/447e30e2c9b24f4ec6af5520a37a77068e66383b/src/firejail/fs.c#L1251-L1299

Done, now it works properly without any extra tweaking.

**Comment by lxeiqr** [comment]
Done, now it works properly without any extra tweaking.

File: src/firejail/fs.c [hunk 2]
```diff
@@ -1274,6 +1274,9 @@ void fs_private_tmp(void) {
 	// read-only x11 directory
 	profile_add("read-only /tmp/.X11-unix");
 
+        // whitelist sndio directory
+        profile_add("whitelist /tmp/sndio");
```
Comment (rusty-snake) [inline src/firejail/fs.c:1278]: 8 spaces -> 1 tab

**Review by rusty-snake — COMMENTED** [review]
LGTM

**Review by rusty-snake — APPROVED** [review]

**Comment by netblue30** [comment]
Thanks for the patch!

**Pull request merged**